addon/validators/base.js

```diff
@@ -89,11 +88,11 @@ const Base = Ember.Object.extend({
 
     if (!isNone(owner)) {
       // Since default error messages are stored in app/validators/messages, we have to look it up via the owner
-      errorMessages = factoryFor(owner, 'validator:messages');
+      errorMessages = owner.factoryFor('validator:messages');
     }
 
     // If for some reason, we can't find the messages object (i.e. unit tests), use default
-    errorMessages = errorMessages || Messages;
+    errorMessages = errorMessages && errorMessages.class ? errorMessages : Messages;
```

Comment:
Once my [PR](https://github.com/rwjblue/ember-factory-for-polyfill/pull/6) into the polyfill is versioned, this line can be reverted.